node/cn/sync.go

```diff
@@ -182,8 +182,8 @@ func (pm *ProtocolManager) getSyncMode(currentBlock *types.Block) downloader.Syn
 
 // synchronise tries to sync up our local block chain with a remote peer.
 func (pm *ProtocolManager) synchronise(peer Peer) {
-	// Short circuit if no peers are available
-	if peer == nil {
+	// Short circuit if no peers are available or syncStop flag is set to true
+	if peer == nil || pm.syncStop {
```

Comment:
same here. Don't we need to use atomic operation?

Reply:
Changed atomically, too.